src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts

```diff
@@ -1896,26 +1900,62 @@ export class SettingsEditor2 extends EditorPane {
 			return null;
 		}
 
+		this.stopWatch.reset();
 		const result = await aiSearchProvider.getLLMRankedResults(token);
-		if (!result || token.isCancellationRequested) {
+		this.stopWatch.stop();
+
+		if (token.isCancellationRequested) {
 			return null;
 		}
 
+		// Only log the elapsed time if there are actual results.
+		if (result && result.filterMatches.length > 0) {
+			const elapsed = this.stopWatch.elapsed();
+			this.logSearchPerformance(LLM_RANKED_SEARCH_PROVIDER_NAME, elapsed);
+		}
+
 		this.searchResultModel!.setResult(SearchResultIdx.AiSelected, result);
 		return result;
 	}
 
-	private async searchWithProvider(type: SearchResultIdx, searchProvider: ISearchProvider, token: CancellationToken): Promise<ISearchResult | null> {
+	private async searchWithProvider(type: SearchResultIdx, searchProvider: ISearchProvider, providerName: string, token: CancellationToken): Promise<ISearchResult | null> {
+		this.stopWatch.reset();
 		const result = await this._searchPreferencesModel(this.defaultSettingsEditorModel, searchProvider, token);
+		this.stopWatch.stop();
+
 		if (token.isCancellationRequested) {
 			// Handle cancellation like this because cancellation is lost inside the search provider due to async/await
 			return null;
 		}
+
+		// Only log the elapsed time if there are actual results.
+		if (result && result.filterMatches.length > 0) {
```

Comment:
I have added this if statement to be consistent with the one for LLM_RANKED results.
It is probably safe to remove the if statement for local and TF-IDF search providers and to keep it for the less available embeddings search provider, but that complicates the if statement and could alter the reported times.